### PR TITLE
Refactor nssm resource 

### PR DESCRIPTION
**Pull request by Annih**
Following last nssm bump with the cookbook v3.0.1 the escaping behavior changed and broke contingent cookbooks. It also added extra execution time to check parameters values.

## Non goals
This PR does not aim to be backward compatible.

## Main changes
* **[breaking]** Handle escaping for the user:
  * No need to use `"""` for parameters with spaces 
  * What the user set in the property is what he gets in nssm.
* Use latest way to define resources (custom resource):
  * Smaller code
  * Separated Platform implementation
* Use version in nssm path to allow upgrade scenario
* Speed up the convergence:
  * Load current service state at once by implementing the `load_current_value`  using `nssm dump`
  * Requires `nssm >= 2.24.94` for the `nssm dump` command
* Introduce `start` & `stop` actions to manipulate the service

The README and tests have been updated accordingly. I did many commits to help see how I came to the current result.

Let me kow if you have any question, remark or feedback.

*Cc.* @aboten

### CODE REVIEW

File: resources/service.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+provides :nssm_service, platform: 'windows'
+# TODO: migrate to nssm_service with a breaking change notice
+provides :nssm, platform: 'windows'
+
+property :servicename, name_attribute: true
+property :program, kind_of: String, required: true
+property :args, kind_of: String
+property :parameters, kind_of: Hash, default: lazy { ::Mash.new }
+property :nssm_binary, kind_of: String, default: lazy { "#{node['nssm']['install_location']}\\nssm.exe" }
+# TODO: migrate this to :start action with a breaking change
```
Comment (dhoer) [inline resources/service.rb]: Should these just be removed if we do a major rev?

File: resources/service.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
+provides :nssm_service, platform: 'windows'
+# TODO: migrate to nssm_service with a breaking change notice
```
Comment (Annih) [inline resources/service.rb]: Yes I think a v4.0.0 would be great, but currently I think my PR is not enough. Due to the way nssm binary is installed upgrade is not possible when a service is currently running.
In addition, the implementation I made for the `load_current_value` does not work with older version of `nssm`, another blocking point for the upgrade.

The `nssm_service` would be great if we start having a resource for `nssm_install` for instance.
I really think we should migrate the `install` recipe into a resource, to enhance install/upgrade scenarii and allow people to subscribe to that kind of event.

File: Gemfile [hunk 3]
```diff
@@ -4,7 +4,7 @@ source 'https://rubygems.org'
 
 gem 'berkshelf'
 gem 'chef', '>= 12.7'
```
Comment (Annih) [inline Gemfile]: Yes maybe we should bump chef in the Gemfile.

**Comment by Annih** [comment]
@dhoer I managed to get green CI :D
I also updated the code to take into account your comments and try to be as much backward-compatible as possible.

I'm creating smaller reviews for different stuffs.

**Comment by Annih** [comment]
Wow quick merge, thanks!

**Comment by dhoer** [comment]
Getting the following error:
resource execute[Install service name service] is configured to notify resource service[service name] with action start, but service[service name] cannot be found in the resource collection. execute[Install service name service] is defined in C:/Users/TEST_U~1/AppData/Local/Temp/kitchen/cache/cookbooks/nssm/resources/service.rb:31:in `block in class_from_file'

Do you have time to look at this?

**Comment by Annih** [comment]
This is related to #22, I was looking at it but your merged it before I managed to get my test OK :D

I'll try again ;)

**Pull request merged**